src/callbacks/onPreprocess/flattenData/calculatePalt.js

```diff
@@ -0,0 +1,77 @@
+export default function calculatePalt(pt) {
+    // Calculates the pAlt value for the given participant ID
+    // For more on PAlt see the following paper: A Rapid Method to Estimate Hepatocyte Loss Due to Drug-Induced Liver Injury by Chung et al
+    // Requires: Baseline visit
+    // Assumes: Units on Alt are IU/L
+
+    var config = this.config;
+
+    //Get a list of raw post-baseline ALT values
+    var alt_values = pt.values.raw
+        .filter(f => f.analysisFlag)
+        .filter(f => f[config.measure_col] == config.measure_values.ALT)
+        .map(function(d) {
+            var obj = {};
+            obj.value = d[config.value_col];
+            obj.day = d[config.studyday_col];
+            obj.analysisFlag = d.analysisFlag;
+            obj.hour = d.day * 24;
+            return obj;
+        });
+
+    if (alt_values.length > 1) {
+        //get peak alt value
+        const alt_peak = d3.max(alt_values, f => f.value);
+
+        //caluculate ALT AUC
+        const alt_auc = d3.sum(alt_values, function(d, i) {
+            if (i < alt_values.length - 1) {
+                const di = d;
+                const vi = di.value;
+                const hi = di.hour;
+                const dii = alt_values[i + 1];
+                const vii = dii.value;
+                const hii = dii.hour;
+                const v_avg = (vii + vi) / 2;
+                const h_diff = hii - hi;
+                const segment_auc = v_avg * h_diff;
+
+                return segment_auc;
+            } else {
+                return 0;
+            }
+        });
+
+        //calculate pAlt  (ALT AUC + AltPeak ^ 0.18)
+        const p_alt = (alt_auc + Math.pow(alt_peak, 0.18)) / 100000;
```

Comment:
i _think_ it's `ALT AUC * AltPeak ^ 0.18`

Reply:
Wow - good catch. Can't believe I messed that up. 😱